Core/ImageReader.cpp

```diff
@@ -59,7 +59,7 @@ bool ImageReader::getInfo2D(const char* filename, unsigned& width, unsigned& hei
 	return ImageReader::getInfo(filename, width, height, nrslices, spacing, tr);
 }
 
-bool ImageReader::getImage2D(const char* filename, float* img, unsigned width, unsigned height, const std::function<float(unsigned char, unsigned char, unsigned char)>& color2grey)
+bool ImageReader::getImageStack(const std::vector<const char*>& filenames, float** img_stack, unsigned width, unsigned height, const std::function<float(unsigned char, unsigned char, unsigned char)>& color2grey)
```

Comment:
std::vector<std::string> instead of vector<const char*>

Reply:
returning a vector of smart unique pointer of images instead of a pointer of pointer of floats seems more secure

Reply:
what comes out when you just define __unsigned__ as a type? is this automatically int?

Reply:
unsigned = unsigned int

Reply:
I am passing the memory, which was already allocated, as vector of pointers -> it is an input.

Reply:
I agree with all your comments, but did not want to change the API already used all over the place.